precompiles/conviction-voting/src/lib.rs

```diff
@@ -182,27 +232,43 @@ where
 		track_id: u16,
 		poll_index: u32,
 	) -> EvmResult {
-		let class = Self::u16_to_track_id(track_id).in_field("trackId")?;
-		let index = Self::u32_to_index(poll_index).in_field("pollIndex")?;
-
-		let target = Runtime::AddressMapping::into_account_id(target.into());
-		let target: <Runtime::Lookup as StaticLookup>::Source =
-			Runtime::Lookup::unlookup(target.clone());
-
-		log::trace!(
-			target: "conviction-voting-precompile",
-			"Removing other vote from poll {:?}",
-			index
-		);
-
-		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
-		let call = ConvictionVotingCall::<Runtime>::remove_other_vote {
-			target,
-			class,
-			index,
-		};
-
-		RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
+		let caller = handle.context().caller;
+		{
+			let class = Self::u16_to_track_id(track_id).in_field("trackId")?;
+			let index = Self::u32_to_index(poll_index).in_field("pollIndex")?;
+
+			let target = Runtime::AddressMapping::into_account_id(target.into());
+			let target: <Runtime::Lookup as StaticLookup>::Source =
+				Runtime::Lookup::unlookup(target.clone());
+
+			log::trace!(
+				target: "conviction-voting-precompile",
+				"Removing other vote from poll {:?}",
+				index
+			);
+
+			let origin = Runtime::AddressMapping::into_account_id(caller);
+			let call = ConvictionVotingCall::<Runtime>::remove_other_vote {
+				target,
+				class,
+				index,
+			};
+
+			RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
+		}
```

Comment:
Moving the old code into an inner scope makes it hard to see the diff and check exactly what changed. 

Probably nothing changed in any of them but it is hard to tell.

Reply:
Right, it's a tradeoff, is that or renaming variables, which one do you prefer?

Reply:
Removed the code block when moving the log record costs to top